### PR TITLE
Tr 33 saving report mvp

### PR DESCRIPTION
**Pull request by DavidNic11**

### CODE REVIEW

File: public/react/SaveReports/services/reports/useRecentReports.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import { Dispatch, SetStateAction, useState } from "react";
+
+const useLocalStorage = <TData = string>(
```
Comment (DavidNic11) [inline public/react/SaveReports/services/reports/useRecentReports.ts:3]: will seperate hooks out later, still messing with them

File: public/shared/state-storage.js [hunk 2]
```diff
@@ -2,109 +2,107 @@ import { RoutePushstate, route, diff } from "../can.js";
...
+    },
+  };
 }
 
 const underlyingReplaceState = history.replaceState;
 
-
 export const pushStateObservable = new RoutePushstate();
-route.urlData = new RoutePushstate();
+route.urlData = pushStateObservable;
```
Comment (DavidNic11) [inline public/shared/state-storage.js:19]: this is the only change in this file

File: public/react/SaveReports/components/SavedReportDropdown/RecentReports.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  return (
+    <DropdownItem
+      key={reportId}
+      onClick={(event) => {
+        window.location.search = "?" + matched.queryParams;
+        // needed to not interfere with can routing
+        event.stopPropagation();
+      }}
+    >
+      Report name {matched.name}
```
Comment (binaryberserker) [inline public/react/SaveReports/components/SavedReportDropdown/RecentReports.tsx:66]: is the dropdown supposed to have "Report name" in the title?

**Comment by binaryberserker** [comment]
getting a scrollbar on the side of the reports page, shrunk my screen for this pic, but i was getting it with the window maximized too (betting it's coming from the main page and probly a pain to fix, not gona hold up this PR for it, just wanted to mention it) 
image

**Comment by binaryberserker** [comment]
the TR-33 AC mentions search on the saved reports page but i don't see anything for that, did it get dropped?

**Comment by binaryberserker** [comment]
only other thing is tests, is it feasible to add any tests for this?

**Pull request merged**